src/weather_model_graphs/networkx_utils.py

```diff
@@ -98,40 +98,29 @@ def split_graph_by_edge_attribute(graph, attr):
             f"No subgraphs were created. Check the edge attribute '{attr}'."
         )
 
-    # copy node attributes
-    for subgraph in subgraphs.values():
-        for node in subgraph.nodes:
-            subgraph.nodes[node].update(graph.nodes[node])
-
-    # check that at least one subgraph was created
-    if len(subgraphs) == 0:
-        raise ValueError(
-            f"No subgraphs were created. Check the edge attribute '{attr}'."
-        )
+    return subgraphs
 
-    # copy node attributes
-    for subgraph in subgraphs.values():
-        for node in subgraph.nodes:
-            subgraph.nodes[node].update(graph.nodes[node])
 
-    # check that at least one subgraph was created
-    if len(subgraphs) == 0:
-        raise ValueError(
-            f"No subgraphs were created. Check the edge attribute '{attr}'."
-        )
+def sort_nodes_in_graph(graph):
+    """
+    Creates a new networkx.DiGraph that is a copy of input, but with nodes
+    sorted according to their id
```

Comment:
```suggestion
    sorted according to their label value
```

I think strictly speaking nodes are identified by something called their "label" rather than "id" (e.g. https://networkx.org/documentation/stable/reference/generated/networkx.relabel.convert_node_labels_to_integers.html#networkx.relabel.convert_node_labels_to_integers). And in our case we use integers as labels